### ISSUE TITLE
Something wrong in i3blocks/scripts/volume?

### ISSUE THREAD

**Issue by cmoang** [vivien/i3blocks#145]
Is something wrong in line 34 of https://github.com/vivien/i3blocks/blob/master/scripts/volume ? I get the same output whether I am using sed or not?

    # amixer -D $MIXER scontrols
    Simple mixer control 'IEC958',0
    Simple mixer control 'IEC958',1
    Simple mixer control 'IEC958',2
    # amixer -D $MIXER scontrols | sed  "s/Simple mixer control '\([A-Za-z ]*\)',0/\1/p" 
    Simple mixer control 'IEC958',0
    Simple mixer control 'IEC958',1
    Simple mixer control 'IEC958',2

When I run `volume` I get this error:

    # ./volume
    Specify a scontrol identifier: 'name',index
    Specify a scontrol identifier: 'name',index

**Comment by Nycroth** [comment]
Ah, it's because the number is staticly expected to 0...

Change the sed expression to `sed  "s/Simple mixer control '\([A-Za-z ]*\)'.*/\1/p"` and everything should work. I'll submit a pull request to fix this.

**Comment by vivien** [comment]
If @cmoang confirms the fix, please issue a PR, thanks ;-)

**Comment by cmoang** [comment]
Wow you are really fast.

tl;dr: for the script to work, one must have installed `extra/pulseaudio-alsa` in Arch.

I have just found out that my system was the culprit and not the volume script. I am on a fresh Arch linux install and the original volume script runs just fine, ***after*** installing the `extra/pulseaudio-alsa` (which for me has the dependency of `alsa-plugins`, too).


So now running `amixer` I get:

    $ amixer
     Simple mixer control 'Master',0
       Capabilities: pvolume pswitch pswitch-joined
       Playback channels: Front Left - Front Right
       Limits: Playback 0 - 65536
       Mono:
       Front Left: Playback 39794 [61%] [off]
       Front Right: Playback 39794 [61%] [off]
     Simple mixer control 'Capture',0
       Capabilities: cvolume cswitch cswitch-joined
       Capture channels: Front Left - Front Right
       Limits: Capture 0 - 65536
       Front Left: Capture 65536 [100%] [on]
       Front Right: Capture 65536 [100%] [on]

while previously poor volume script had to deal with:

    $ amixer
     Simple mixer control 'IEC958',0
       Capabilities: pswitch pswitch-joined
       Playback channels: Mono
       Mono: Playback [off]
     Simple mixer control 'IEC958',1
       Capabilities: pswitch pswitch-joined
       Playback channels: Mono
       Mono: Playback [off]
     Simple mixer control 'IEC958',2
       Capabilities: pswitch pswitch-joined
       Playback channels: Mono
       Mono: Playback [off]
    
Specifically, what is required is this in `/etc/asound.conf`:

    ctl.!default {
      type pulse
    }

So now everything is fine. Sorry for the hiccup...
